src/coreclr/tools/Common/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -37,6 +37,13 @@ public enum RelocType
         // Windows x64
         IMAGE_REL_SECREL                     = 0x104,
 
+        // Windows arm64 TLS access
+        // The value of IMAGE_REL_BASED_ARM64_SECREL_HIGH12A and IMAGE_REL_BASED_DIR64 are same.
```

Comment:
I am not following. Where is `IMAGE_REL_BASED_ARM64_SECREL_HIGH12A` defined to be same as `IMAGE_REL_BASED_DIR64`?

Reply:
It is mapped in `CoffObjectWriter.cs` https://github.com/dotnet/runtime/pull/104282/files#diff-00ac2ca96470f5f4a276ac0beeccd215c78dda3bbccf8de38eeb73c2da8ac3a8R357-R358

Reply:
`IMAGE_REL_ARM64_SECREL_HIGH12A` (without _BASED) happens to have the same value as `IMAGE_REL_BASED_DIR64`. `IMAGE_REL...` and `IMAGE_REL_BASED_...` are two different enumerations that are not interchangeable. It would expect that number of other values overlap in the same way.

(`IMAGE_REL_BASED_,,,,` enumerations is our own somewhat broken invention. It would be more appropriate to use `IMAGE_REL_ARM64_SECREL_` instead, but that would be a separate refactoring.)

Reply:
I think these values should be after Linux arm32 ones and follow the sequence. I would omit this comment.

Reply:
> It would be more appropriate to use IMAGE_REL_ARM64_SECREL_ instead, but that would be a separate refactoring.)

I started with `IMAGE_REL_ARM64_SECREL_HIGH12A` and `IMAGE_REL_BASED_ARM64_SECREL_LOW12L`, but those names overlap with the ones in `CoffRelocationType` and I get build errors, that's why added `BASED` in it. If you want I can change it to something like `IMAGE_REL_ARM64_SECREL_TLS_HIGH12A` and `IMAGE_REL_BASED_ARM64_SECREL_TLS_LOW12L` or anything else meaningful.

```
Ambiguity between 'CoffObjectWriter.CoffRelocationType.IMAGE_REL_ARM64_SECREL_HIGH12A' and 'RelocType.IMAGE_REL_ARM64_SECREL_HIGH12A'	ILCompiler.Compiler	D:\git\runtime\src\coreclr\tools\aot\ILCompiler.Compiler\Compiler\ObjectWriter\CoffObjectWriter.cs	357	
```

> I think these values should be after Linux arm32 ones and follow the sequence. I would omit this comment.

Any particular reason? If we want to order it by the numbers, they should be towards beginning. If I should place it after linux/arm32, probably, I should have their values to something like `0x20A` and `0x20B`.

Reply:
When I was adding the TLS entries to this enum I only cared about entries for the same OS to be together, otherwise I was just adding sequentially and assigning the numbers sequentially. These are just number that we later remap. They cannot match the actual reloc value since reloc values are contextual on OS and target instruction and thus often overlap.

There is some esthetical thing in win-arm64 entries going after win-x64, but that means - either values are not sequential, or you'd have to shift all the subsequent values. 

Perhaps just continue the sequence after Linux arm32 and assigning the next values in sequence is the simplest. 

The order in the enum will basically be in the historical implementation order.

Reply:
Sounds good. What about the names of the new enums? Should it have `BASED` or something like `IMAGE_REL_ARM64_SECREL_TLS_HIGH12A ` and `IMAGE_REL_BASED_ARM64_SECREL_TLS_LOW12L `?

Reply:
I have small preference for the name without `_BASED_`.

Reply:
sure, will push that change once `superpmi-diff` results come out. I have to update guid as well anyway.

Reply:
Done.